<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SyncConfigDialog</class>
 <widget class="QDialog" name="SyncConfigDialog">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>459</width>
    <height>326</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sync Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="styleSheet">
          <string notr="true">font-size: 16px;</string>
         </property>
         <property name="text">
          <string>Symphytum is able to synchronize your database including all data with a cloud service, allowing you to keep data safe and available everywhere on all devices.</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Cloud service:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="serviceComboBox"/>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="cancelButton">
           <property name="text">
            <string>&amp;Cancel</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="loginButton">
           <property name="text">
            <string>&amp;Login</string>
           </property>
           <property name="default">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="oauthConfig">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="urlLabelOK">
         <property name="styleSheet">
          <string notr="true">font-size: 16px;</string>
         </property>
         <property name="text">
          <string>Please authorize access to your cloud service: &lt;br /&gt; copy the code from the opened browser window and paste it below</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="urlLabelResult">
         <property name="styleSheet">
          <string notr="true">font-size: 16px;</string>
         </property>
         <property name="text">
          <string>Please wait...</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="urlLabel">
           <property name="text">
            <string>Requesting URL...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QProgressBar" name="urlProgressBar">
           <property name="maximum">
            <number>0</number>
           </property>
           <property name="value">
            <number>-1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="codeLabel">
           <property name="text">
            <string>&amp;Code:</string>
           </property>
           <property name="buddy">
            <cstring>codeLineEdit</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="codeLineEdit">
           <property name="placeholderText">
            <string>Paste here the code from the browser window</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="cancelUrlButton">
           <property name="text">
            <string>&amp;Cancel</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="okButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>&amp;OK</string>
           </property>
           <property name="default">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_3">
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="resultLabel">
         <property name="styleSheet">
          <string notr="true">font-size: 16px;</string>
         </property>
         <property name="text">
          <string>Please wait...</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
         <property name="openExternalLinks">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="authLabel">
           <property name="text">
            <string>Authorizing...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QProgressBar" name="authProgressBar">
           <property name="maximum">
            <number>0</number>
           </property>
           <property name="value">
            <number>-1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>90</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="cancelAuthButton">
           <property name="text">
            <string>&amp;Cancel</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="retryButton">
           <property name="text">
            <string>&amp;Retry</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="finishButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>&amp;Finish</string>
           </property>
           <property name="default">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="megaConfig">
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QLabel" name="megaConfLabel">
         <property name="styleSheet">
          <string notr="true">font-size: 16px;</string>
         </property>
         <property name="text">
          <string>Please authorize access to your MEGA account</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>MEGA email:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="megaEmailLineEdit">
           <property name="placeholderText">
            <string>name@example.com</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>MEGA password:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="megaPassLineEdit">
           <property name="echoMode">
            <enum>QLineEdit::Password</enum>
           </property>
           <property name="placeholderText">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox">
         <property name="toolTip">
          <string>Enable this if your MEGA account is protected by 2FA. &lt;br /&gt; If you don't know what this is then your account is probably not using 2FA.</string>
         </property>
         <property name="text">
          <string>Use two factor authentication (2FA)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="mega2FAGroupBox">
         <property name="title">
          <string>2FA</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Two factor code (2FA):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="mega2FALineEdit">
            <property name="placeholderText">
             <string>(optional, use if 2FA is enabled)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>&lt;i&gt;Your password will not be stored, only a session key&lt;/i&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_7">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>58</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="Line" name="line_7">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_17">
         <item>
          <spacer name="horizontalSpacer_9">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="cancelMegaButton">
           <property name="text">
            <string>&amp;Cancel</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="okMegaButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>&amp;OK</string>
           </property>
           <property name="default">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>serviceComboBox</tabstop>
  <tabstop>loginButton</tabstop>
  <tabstop>cancelButton</tabstop>
  <tabstop>okButton</tabstop>
  <tabstop>cancelUrlButton</tabstop>
  <tabstop>finishButton</tabstop>
  <tabstop>cancelAuthButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>codeLineEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>okButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>89</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>megaPassLineEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>okMegaButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>376</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>megaEmailLineEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>okMegaButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mega2FALineEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>okMegaButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>367</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>346</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>mega2FAGroupBox</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>39</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>52</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
